.github/workflows/sync_from_airtable.yml

```diff
@@ -0,0 +1,330 @@
+name: Sync from Airtable
+
+on:
+  repository_dispatch:
+    types: [airtable-update]
+
+env:
+  DATA: ${{ toJSON(github.event.client_payload.data) }}
+  NUMBER: ${{ github.event.client_payload.number }}
+  TITLE: ${{ github.event.client_payload.title }}
+  OWNER: "Esri"
+  REPO: "calcite-components"
+
+jobs:
+  existing_issue:
+    name: Find Existing Issue
+    runs-on: ubuntu-latest
+    outputs:
+      github_issue: ${{ steps.find-existing-issue.outputs.result }}
+    steps:
+      - uses: actions/github-script@v6
+        id: find-existing-issue
+        env:
+          NUMBER: "${{env.NUMBER}}"
+          OWNER: "${{env.OWNER}}"
+          REPO: "${{env.REPO}}"
+          TITLE: "${{env.TITLE}}"
+          ACTION: "${{github.event.client_payload.action}}"
+        with:
+          result-encoding: string
+          script: |
+            const { NUMBER, OWNER, REPO, ACTION, TITLE } = process.env;
+            try {
+                const issue = await github.rest.issues.get({
+                    issue_number: NUMBER,
+                    owner: OWNER,
+                    repo: REPO,
+                });
+
+                return issue.status === 200 ? JSON.stringify(issue) : JSON.stringify(null);
+            } catch (error) {
+                
+                const issue = await github.rest.search.issuesAndPullRequests({
+                    q: `"${TITLE}" in:title is:open is:issue`,
+                    sort: 'created',
+                    order: 'asc',
+                    per_page: 1
+                }).then((result) =>  result[0]);
+                core.info(issue);
+
+                return issue ? JSON.stringify(issue) : JSON.stringify(null);
+            }
+
+            return JSON.stringify(null);
+
+  assigned_unassigned:
+    name: Assign or Unassign
+    needs: existing_issue
+    if: (fromJSON(needs.existing_issue.outputs.github_issue).status == 200 ) && (github.event.client_payload.action == 'assigned/unassigned')
```

Comment:
these are custom actions being sent by Airtable so I have complete control over the payload. On the Airtable side I'm listening for the default Github action expressions but when being sent back I'm combining them because the behavior is the same.